### PR TITLE
refactor(kubernetes): convert deploy manifest stage to react

### PR DESCRIPTION
**Pull request by maggieneterval**
- Adds `StageArtifactSelectorDelegate` which conditionally renders `StageArtifactSelector` or `PreRewriteStageArtifactSelector` depending on `artifactsRewrite` feature flag.
- Adds `ManifestBindArtifactsSelectorDelegate` which conditionally renders `ManifestBindArtifactsSelector` or new `PreRewriteManifestBindArtifactsSelector` depending on `artifactsRewrite` feature flag.
- Changes `PreRewriteStageArtifactSelector` prop key from `excludedArtifactTypes` to `excludedArtifactTypePatterns` for consistency with `StageArtifactSelector` props. Updates this key in consumers (`GoogleCloudBuildStageForm` and `KubernetesV2RunJobStage`).
- Renames `deployManifest/react` directory, which contains components for displaying the status of a manifest, to `deployManifest/manifestStatus`.
- Removes now-unused Angular components: `basicSettings.component`, `deployManifestConfig.controller`, `deployManifestConfig.html`
- Removes Angular dependencies from `CopyFromTemplateButton`, `ManifestDeploymentOptions`, `ManifestBindArtifactsSelector`, `ManifestCopier`
- Style change: moves "copy from running infrastructure" link from next to "Text" radio button to above YAML editor, replaces pre-rewrite "Required Artifacts to Bind" Angular multi-select with `react-select` multi-select, moves YAML editor inline with other config fields rather than full screen (feedback welcome on any of these changes).
- artifactsRewrite=false, source=text:
preRewriteText
- artifactsRewrite=false, source=artifact:
z53RmRdeVA4
- artifactsRewrite=true, source=artifact (inline):
5t9k1wD2A8V

### CODE REVIEW

File: app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/PreRewriteManifestBindArtifactSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  onChangeBindings: (bindings: IManifestBindArtifact[]) => void;
+}
+
+export class PreRewriteManifestBindArtifactSelector extends React.Component<IExpectedArtifactMultiSelectorProps> {
+  private getAvailableExpectedArtifacts = (): IExpectedArtifact[] => {
+    return ExpectedArtifactService.getExpectedArtifactsAvailableToStage(this.props.stage, this.props.pipeline);
+  };
+
+  private getOptions = (): Array<Option<string>> => {
+    return map(this.getAvailableExpectedArtifacts(), a => ({
```
Comment (louisjimenez) [inline app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/PreRewriteManifestBindArtifactSelector.tsx]: nit: curious as to why lodash `map` over native here

Comment (maggieneterval) [reply app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/PreRewriteManifestBindArtifactSelector.tsx]: no good reason, just prettier :nail_care:

Comment (maggieneterval) [reply app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/PreRewriteManifestBindArtifactSelector.tsx]: (and null-safe)

**Comment by maggieneterval** [comment]
Thanks all for the code review, I know this was a long one :sweat_smile: 
cc @jkschneider the delegate components I added here might be useful for any future React stage conversions that need to be compatible with pre and post artifact rewrite world.

**Pull request merged**